### PR TITLE
mgr/dashboard: Introduce SASS modules

### PR DESCRIPTION
**Pull request by SchoolGuy**
<!--
This is the format for commit messages:

"""
[component]: [short description]

[A longer multiline description]

Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
Signed-off-by: [Your Name] <[your email]>
"""
-->
This PR is trying to achieve two things:

- Introduce SASS modules to the internal SASS structure of Ceph
- Do some cleanup/structuring to the .scss files we have. Thus also extending the `bootstrap-extend.scss` approach until upstream Bootstrap has module support.

This change is very intrusive and thus needs to be reviewed and revised properly.

## Statistics

I am generating this with:
```
npm run env_build & ng build --prod --stats-json --configuration=$npm_package_config_locale
```

I am running my statistics on a machine with 32GB RAM (3000 MHz) and a Ryzen 2700X (not overclocked). Code is living on an Samsung 860 QVO.

### Code in this PR

```
╭─root@new-ceph-container / 
╰─# npm run-script build

> ceph-dashboard@0.0.0 build /ceph/src/pybind/mgr/dashboard/frontend
> npm run env_build & ng build --prod --stats-json --configuration=$npm_package_config_locale


> ceph-dashboard@0.0.0 env_build /ceph/src/pybind/mgr/dashboard/frontend
> node cd --env

[cd.js] 'src/environments/environment.tpl.ts' was copied to 'src/environments/environment.prod.ts'
[cd.js] 'src/environments/environment.tpl.ts' was copied to 'src/environments/environment.ts'
[cd.js] Placeholders were replace in 'src/environments/environment.prod.ts'
[cd.js] Placeholders were replace in 'src/environments/environment.ts'

chunk {} runtime.dbd1feac31f9eb48da56.js (runtime) 2.28 kB [entry] [rendered]
chunk {1} 1.cc9360ff6156a0807125.js () 232 kB  [rendered]
chunk {2} main.d8842fc8905b6819e8e1.js (main) 5.13 MB [initial] [rendered]
chunk {3} polyfills.0f5bf1f63d3d2dd733ef.js (polyfills) 36.2 kB [initial] [rendered]
chunk {4} styles.886a4614942911d4cff0.css (styles) 207 kB [initial] [rendered]
chunk {5} 5.517b115eb0b7203d9056.js () 91.3 kB  [rendered]
chunk {6} 6.15501b61a6e5e1bc0cc1.js () 111 kB  [rendered]
chunk {scripts} scripts.38cee5fbd56812617717.js (scripts) 205 kB [entry] [rendered]
Date: 2020-06-09T14:02:11.953Z - Hash: 7ffb6e1eb7e73fd46369 - Time: 55346ms


Localized bundle generation complete.
```

### Current master branch

```
╭─root@new-ceph-container / 
╰─# npm run-script build                                                                                                                                                                                                               130 ↵

> ceph-dashboard@0.0.0 build /ceph/src/pybind/mgr/dashboard/frontend
> npm run env_build & ng build --prod --stats-json --configuration=$npm_package_config_locale


> ceph-dashboard@0.0.0 env_build /ceph/src/pybind/mgr/dashboard/frontend
> node cd --env

[cd.js] 'src/environments/environment.tpl.ts' was copied to 'src/environments/environment.ts'
[cd.js] 'src/environments/environment.tpl.ts' was copied to 'src/environments/environment.prod.ts'
[cd.js] Placeholders were replace in 'src/environments/environment.ts'
[cd.js] Placeholders were replace in 'src/environments/environment.prod.ts'

chunk {} runtime.287f3de405b483f66417.js (runtime) 2.28 kB [entry] [rendered]
chunk {1} 1.bfd7af2316e653845713.js () 232 kB  [rendered]
chunk {2} main.f1a9ec35abc4155c1573.js (main) 5.13 MB [initial] [rendered]
chunk {3} polyfills.0f5bf1f63d3d2dd733ef.js (polyfills) 36.2 kB [initial] [rendered]
chunk {4} styles.e3319288f37b2cc63988.css (styles) 207 kB [initial] [rendered]
chunk {5} 5.c790c51c88ca819465aa.js () 91.3 kB  [rendered]
chunk {6} 6.d96cae8cdf49a12c4652.js () 111 kB  [rendered]
chunk {scripts} scripts.38cee5fbd56812617717.js (scripts) 205 kB [entry] [rendered]
Date: 2020-06-09T13:53:08.138Z - Hash: 55db778986d490b152da - Time: 86573ms


Localized bundle generation complete.
```

## TODO-List

- [x] Cleanup Git History
- [x] Provide statistics for compilation times and webpack size
- [x] Look for candidates which can be merged into `bootstrap-extends.scss` (Postponed to new PR)

## Checklist
- [x] References tracker ticket: https://tracker.ceph.com/issues/45934
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by SchoolGuy** [comment]
I force pushed to remove the conflicts and redid the PR again. Thus now only one commit because it is a single step.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard/health/health.component.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@import 'defaults.scss';
+@use './src/styles/defaults/bootstrap-defaults';
```
[the remark this answers is not in the document]
Comment (SchoolGuy) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard/health/health.component.scss:1]: Yes that alias is certainly possible.

File: src/pybind/mgr/dashboard/frontend/src/styles/ceph-custom/_buttons.scss [hunk 2]
```diff
@@ -0,0 +1,64 @@
+@use "../defaults";
```
[the remark this answers is not in the document]
Comment (SchoolGuy) [reply src/pybind/mgr/dashboard/frontend/src/styles/ceph-custom/_buttons.scss:1]: I think we should just fix this file. Then it is more explicit from where the styles are coming from. Regarding importing folders vs single files. That is a SASS Module thing I recommend to read this https://sass-lang.com/documentation/at-rules/use#index-files
(My explaining isn't the best so I just pasted this here.)

**Comment by Exotelis** [comment]
jenkins test dashboard

**Comment by SchoolGuy** [comment]
I rebased wrong. I haven't kept my main `_bootstrap-defaults.scss` file. I will recover that tomorrow. DNM until then!

File: src/pybind/mgr/dashboard/frontend/src/styles/ceph-custom/_buttons.scss [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  &:focus:not(:checked) ~ .custom-control-label::before {
+    border-color: $custom-control-indicator-focus-border-color;
+  }
+}
+
+/* Buttons */
+.btn-light {
+  background-color: bd.$color-solid-white !important;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/styles/ceph-custom/_buttons.scss:30]:
```suggestion
  background-color: bd.$color-solid-white;
```

File: src/pybind/mgr/dashboard/frontend/src/styles/ceph-custom/_buttons.scss [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  border-color: bd.$color-input-border !important;
+
+  &:hover {
+    background-color: bd.$color-soft-gray !important;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/styles/ceph-custom/_buttons.scss:34]:
```suggestion
    background-color: bd.$color-soft-gray;
```

**Comment by SchoolGuy** [comment]
Force push to hopefully fix the tests.

**Comment by SchoolGuy** [comment]
@tspmelo Do you want to have all 4 `!importants` removed or only those marked two?

File: src/pybind/mgr/dashboard/frontend/src/styles/ceph-custom/_buttons.scss (hunk 3 above)
Comment (SchoolGuy) [reply src/pybind/mgr/dashboard/frontend/src/styles/ceph-custom/_buttons.scss]: Done

File: src/pybind/mgr/dashboard/frontend/src/styles/ceph-custom/_buttons.scss (hunk 4 above)
Comment (SchoolGuy) [reply src/pybind/mgr/dashboard/frontend/src/styles/ceph-custom/_buttons.scss]: Done

**Comment by tspmelo** [comment]
@SchoolGuy yes, just the 2, so we have the code as it is in master.

Please run `npm run fix`. This will fix part of the linting errors you are having.

**Comment by SchoolGuy** [comment]
Discussed with @tspmelo: We have the following issue with our linter: https://github.com/sh-waqar/stylelint-declaration-use-variable/issues/37

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard/info-card/info-card-popover.scss [hunk 5]
```diff
@@ -11,32 +11,30 @@
...
   }
 }
 
 .info-card-content-clickable {
   border: 1px solid;
-  border-color: $color-shade-gray;
+  border-color: bd.$color-shade-gray;
   border-radius: 3px;
-  cursor: pointer;
-  padding: 7px;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard/info-card/info-card-popover.scss:36]: This 2 lines seem to be missing.

File: src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.scss [hunk 6]
```diff
@@ -24,36 +24,34 @@
...
-    color: $color-password-toggle-text;
+    background-color: bd.$color-password-toggle-bg;
+    color: bd.$color-password-toggle-text;
   }
 }
 
 input:-webkit-autofill {
   animation-fill-mode: both;
-  animation-name: autofill;
-  border-radius: 0;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.scss:55]: These 2 lines are missing.

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/breadcrumbs/breadcrumbs.component.scss [hunk 7]
```diff
@@ -8,12 +8,14 @@
 }
 
 .breadcrumb > li + li::before {
-  color: $color-breadcrumb;
+  color: bd.$color-breadcrumb;
   content: '\f101';
   font-family: 'ForkAwesome';
+  font-family: 'ForkAwesome';
+  padding: 0 5px 0 7px;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/breadcrumbs/breadcrumbs.component.scss:15]: These are repeated.
```suggestion
```

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.scss [hunk 8]
```diff
@@ -42,24 +42,23 @@
...
-      color: $color-nav-links;
+      color: bd.$color-nav-links;
     }
   }
 
   .navbar-nav > li > .cd-navbar > [ngbDropdown] > a,
   .navbar-nav > li > .cd-navbar > a,
   .navbar-nav > li > a {
-    color: $color-nav-links;
-    display: block;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.scss:62]: This line is missing

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.scss [hunk 9]
```diff
@@ -112,23 +111,22 @@
       }
 
       .open .dropdown-menu {
-        background-color: $color-nav-open-bg;
+        background-color: bd.$color-nav-open-bg;
         border: 0;
         padding-bottom: 0;
-        padding-top: 0;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.scss:118]: This line is missing.

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.scss [hunk 10]
```diff
@@ -112,23 +111,22 @@
...
       }
 
       .open .dropdown-menu > .active > a {
-        background-color: $color-nav-active-link-bg;
+        background-color: bd.$color-nav-active-link-bg;
       }
 
-      > li > a:hover {
-        background-color: $color-nav-active-link-bg;
+      & > li > a:hover {
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.scss:128]:
```suggestion
      > li > a:hover {
```

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.scss [hunk 11]
```diff
@@ -223,7 +221,8 @@ $sidebar-width: 200px;
   }
 
   ul ul a {
-    background: lighten($color-navbar-bg, 10);
+    background: lighten(bd.$color-navbar-bg, 10);
+    background: lighten(bd.$color-navbar-bg, 10);
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.scss:225]:
```suggestion
    background: lighten(bd.$color-navbar-bg, 10);
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/components/select/select.component.scss [hunk 12]
```diff
@@ -1,13 +1,15 @@
-@import 'defaults.scss';
+@use './src/styles/defaults/bootstrap-defaults' as bd;
 
 .select-menu-item {
-  border-bottom: 1px solid $color-transparent;
+  border-bottom: 1px solid bd.$color-transparent;
+  border-bottom: 1px solid bd.$color-transparent;
+  cursor: pointer;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/components/select/select.component.scss:6]:
```suggestion
  border-bottom: 1px solid bd.$color-transparent;
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/components/usage-bar/usage-bar.component.scss [hunk 13]
```diff
@@ -18,10 +18,12 @@
   }
 
   span {
-    color: $color-progress-text;
+    color: bd.$color-progress-text;
     display: block;
     font-weight: normal;
     position: absolute;
+    position: absolute;
+    width: 100%;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/components/usage-bar/usage-bar.component.scss:26]:
```suggestion
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss [hunk 14]
```diff
@@ -11,16 +12,17 @@
   margin-bottom: 25px;
 
   .separator {
-    border-left: 1px solid $color-table-seperator-border;
+    border-left: 1px solid bd.$color-table-seperator-border;
+    border-left: 1px solid bd.$color-table-seperator-border;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss:16]:
```suggestion
    border-left: 1px solid bd.$color-table-seperator-border;
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss [hunk 15]
```diff
@@ -11,16 +12,17 @@
...
     display: inline-block;
     height: 30px;
-    margin-left: 5px;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss:17]: This line is missing.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss [hunk 16]
```diff
@@ -11,16 +12,17 @@
...
     padding-left: 5px;
     vertical-align: middle;
   }
 
   .widget-toolbar {
-    border-left: 1px solid $color-table-seperator-border;
+    border-left: 1px solid bd.$color-table-seperator-border;
+    border-left: 1px solid bd.$color-table-seperator-border;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss:25]:
```suggestion
    border-left: 1px solid bd.$color-table-seperator-border;
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss [hunk 17]
```diff
@@ -36,9 +38,9 @@
 }
 
 .dataTables_header {
-  background-color: $color-table-header-bg;
-  border: 1px solid $color-table-header-border;
-  border-bottom: 0;
+  background-color: bd.$color-table-header-bg;
+  border: 1px solid bd.$color-table-header-border;
+  border-bottom: none;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss:43]:
```suggestion
  border-bottom: 0;
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss [hunk 18]
```diff
@@ -47,7 +49,8 @@
   }
 
   ::ng-deep .table-filters {
-    border-left: 1px solid $color-table-seperator-border;
+    border-left: 1px solid bd.$color-table-seperator-border;
+    border-left: 1px solid bd.$color-table-seperator-border;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss:53]:
```suggestion
    border-left: 1px solid bd.$color-table-seperator-border;
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss [hunk 19]
```diff
@@ -61,7 +64,8 @@
   }
 
   .input-group {
-    border-left: 1px solid $color-table-input-border;
+    border-left: 1px solid bd.$color-table-input-border;
+    border-left: 1px solid bd.$color-table-input-border;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss:68]:
```suggestion
    border-left: 1px solid bd.$color-table-input-border;
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss [hunk 20]
```diff
@@ -125,23 +129,32 @@
...
+    background-image: -webkit-linear-gradient(
+      top,
+      bd.$color-table-gradient-1 0,
+      bd.$color-table-gradient-2 100%
+    );
+    background-image: -o-linear-gradient(
+      top,
+      bd.$color-table-gradient-1 0,
+      bd.$color-table-gradient-2 100%
+    );
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss:142]:
```suggestion
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss [hunk 21]
```diff
@@ -125,23 +129,32 @@
...
 
     .sort-asc,
     .sort-desc {
-      color: $color-table-sort;
+      color: bd.$color-table-sort;
     }
 
     .datatable-header-cell {
-      @include table-cell;
-      font-weight: bold;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss:144]: This line is missing.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss [hunk 22]
```diff
@@ -184,8 +197,7 @@
     margin-bottom: -6px;
 
     .empty-row {
-      background-color: $color-table-empty-row;
-      font-style: italic;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss:188]: This line is missing.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss [hunk 23]
```diff
@@ -194,27 +206,26 @@
 
     .datatable-body-row {
       &.clickable:hover .datatable-row-group {
-        background-color: $color-table-hover-row;
-        transition-duration: 0.3s;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss:198]: This line is missing.

File: src/pybind/mgr/dashboard/frontend/src/styles.scss [hunk 24]
```diff
@@ -494,21 +133,25 @@ bfv-messages {
   }
 
   .weak {
+    background: $color-solid-red;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/styles.scss:136]:
```suggestion
```

File: src/pybind/mgr/dashboard/frontend/src/styles.scss [hunk 25]
```diff
@@ -494,21 +133,25 @@ bfv-messages {
...
     background: $color-solid-red;
     width: 25%;
   }
 
   .ok {
+    background: $color-bright-yellow;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/styles.scss:142]:
```suggestion
```

File: src/pybind/mgr/dashboard/frontend/src/styles.scss [hunk 26]
```diff
@@ -494,21 +133,25 @@ bfv-messages {
...
     background: $color-bright-yellow;
     width: 50%;
   }
 
   .strong {
+    background: $color-bright-green;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/styles.scss:148]:
```suggestion
```

File: src/pybind/mgr/dashboard/frontend/src/styles.scss [hunk 27]
```diff
@@ -494,21 +133,25 @@ bfv-messages {
...
     background: $color-bright-green;
     width: 75%;
   }
 
   .very-strong {
+    background: $color-green;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/styles.scss:154]:
```suggestion
```

File: src/pybind/mgr/dashboard/frontend/src/app/core/auth/login-password-form/login-password-form.component.scss [hunk 28]
```diff
@@ -1,27 +1 @@
...
-  }
-
-  .btn-password:focus {
-    outline-color: $color-password-toggle-focus;
-  }
-
-  button.btn:not(:first-child) {
-    margin-left: 5px;
-  }
-}
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/auth/login-password-form/login-password-form.component.scss:27]: This is missing.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard/info-card/info-card.component.scss [hunk 29]
```diff
@@ -1,20 +1,21 @@
...
     $card-font-min-width,
     $card-font-max-width,
     $card-font-min-size,
     $card-font-max-size
   );
-  border: 0.5px solid $color-info-card-border;
+  border: 0.5px solid bd.$gray-300;
   border-radius: 3px;
-  box-shadow: 0 1px 1px $color-shadow-gray;
+  box-shadow: 0 1px 1px rgba(3, 3, 3, 0.175);
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard/info-card/info-card.component.scss:18]: Please change the color with a `var`.

File: src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.scss [hunk 30]
```diff
@@ -1,59 +1,49 @@
-@import 'defaults.scss';
+@use 'sass:map';
+@use './src/styles/defaults/bootstrap-defaults' as bd;
 
-::ng-deep cd-login {
-  h1 {
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.scss:4]: this is missing

File: src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.scss [hunk 31]
```diff
@@ -1,59 +1,49 @@
...
+    border-left: 4px solid bd.$white;
     height: auto;
     margin-bottom: 2rem;
 
     &:focus-within {
-      border-left: 4px solid $color-accent;
-    }
-
-    .invalid-feedback {
-      display: inline;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.scss:19]: This is missing

File: src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.scss [hunk 32]
```diff
@@ -1,59 +1,49 @@
...
     box-shadow: none;
-    color: $color-password-toggle-text;
+    color: bd.$body-color-bright;
     filter: none;
     outline: none;
   }
 
-  .placeholder {
-    color: $color-password-toggle-placeholder-text;
-  }
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.scss:37]: this is missing.

File: src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.scss [hunk 33]
```diff
@@ -1,59 +1,49 @@
...
-    background-color: $color-password-toggle-bg;
-    color: $color-password-toggle-text;
+    background-color: $dark-secondary;
+    color: bd.$body-color-bright;
   }
 }
 
 input:-webkit-autofill {
   animation-fill-mode: both;
-  animation-name: autofill;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.scss:54]: this is missing.

File: src/pybind/mgr/dashboard/frontend/src/app/core/auth/role-details/role-details.component.scss [hunk 34]
```diff
@@ -1,13 +1,9 @@
-@import 'defaults.scss';
-
-thead {
-  background-color: $color-table-header-bg;
-}
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/auth/role-details/role-details.component.scss:5]: this is missing.

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/breadcrumbs/breadcrumbs.component.scss [hunk 35]
```diff
@@ -8,12 +6,7 @@
 }
 
 .breadcrumb > li + li::before {
-  color: $color-breadcrumb;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/breadcrumbs/breadcrumbs.component.scss:11]: this is missing.

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/breadcrumbs/breadcrumbs.component.scss [hunk 36]
```diff
@@ -8,12 +6,7 @@
...
   content: '\f101';
   font-family: 'ForkAwesome';
   padding: 0 5px 0 7px;
 }
-
-.breadcrumb > li > span {
-  color: $color-breadcrumb;
-}
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/breadcrumbs/breadcrumbs.component.scss:19]: this is missing.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss [hunk 37]
```diff
@@ -125,24 +126,22 @@
...
     background-clip: padding-box;
-    background-color: $color-table-datatable-header;
-    background-image: linear-gradient(
-      to bottom,
-      $color-table-gradient-1 0,
-      $color-table-gradient-2 100%
-    );
+    background-color: bd.$gray-100;
+    background-image: -webkit-linear-gradient(top, bd.$gray-100 0, bd.$gray-200 100%);
+    background-image: -o-linear-gradient(top, bd.$gray-100 0, bd.$gray-200 100%);
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss:131]:
```suggestion
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss [hunk 38]
```diff
@@ -194,27 +193,26 @@
 
     .datatable-body-row {
       &.clickable:hover .datatable-row-group {
-        background-color: $color-table-hover-row;
-        transition-duration: 0.3s;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.scss:198]: this is missing.

File: src/pybind/mgr/dashboard/frontend/src/styles/ceph-custom/_dropdown.scss [hunk 39]
```diff
@@ -0,0 +1,35 @@
...
+  z-index: 999999;
+
+  button.dropdown-item:focus {
+    outline: none;
+  }
+}
+
+.dropdown-menu > li > a {
+  cursor: pointer;
+}
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/styles/ceph-custom/_dropdown.scss:15]: color is missing:
```suggestion
.dropdown-menu > li > a {
  color: $color-dropdown-menu;
  cursor: pointer;
}
```

File: src/pybind/mgr/dashboard/frontend/src/styles/ceph-custom/_grid.scss [hunk 40]
```diff
@@ -0,0 +1,5 @@
+@use '../defaults/bootstrap-defaults' as bd;
+
+.container-fluid {
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/styles/ceph-custom/_grid.scss:3]:
```suggestion
.container,
.container-fluid {
```

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test dashboard backend

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by SchoolGuy** [comment]
`make check` fails unrelated to my PR I think.
This is the oposing test: `unittest_rgw_dmclock_scheduler`.

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Pull request merged**